src/MembershipManager.php

```diff
@@ -97,25 +97,20 @@ public function getMemberships(AccountInterface $user, array $states = [OgMember
     ];
     $identifier = implode(':', $identifier);
 
-    // Return cached result if it exists.
-    if (isset($this->cache[$identifier])) {
-      return $this->cache[$identifier];
-    }
-
-    $query = $this->entityTypeManager
-      ->getStorage('og_membership')
-      ->getQuery()
-      ->condition('uid', $user->id())
-      ->condition('state', $states, 'IN');
+    // Use cached result if it exists.
+    if (!isset($this->cache[$identifier])) {
+      $query = $this->entityTypeManager
+        ->getStorage('og_membership')
+        ->getQuery()
+        ->condition('uid', $user->id())
+        ->condition('state', $states, 'IN');
 
-    $results = $query->execute();
+      $this->cache[$identifier] = $query->execute();
+    }
 
-    /** @var \Drupal\og\Entity\OgMembership[] $memberships */
-    $this->cache[$identifier] = $this->entityTypeManager
+    return $this->entityTypeManager
       ->getStorage('og_membership')
-      ->loadMultiple($results);
-
-    return $this->cache[$identifier];
+      ->loadMultiple($this->cache[$identifier]);
```

Comment:
`$this->cache[$identifier]` can also be an empty array at this point, no?

Reply:
Yes good point. In the code before this PR this was also happening, but I will still put a switch in so we don't needlessly try to do a load.

Reply:
I fixed it here in #451 which already refactors this line: https://github.com/Gizra/og/pull/451/commits/84e5553ebc97a7e5763df52076f3ddc729a35345

This saves me a round of rebasing :)